File: Dart/q7.dart
import 'dart:io';

bool isPrime(int number) {
  if (number <= 1) {
    return false;
  }
  for (int i = 2; i <= (number ~/ 2); i++) {
    if (number % i == 0) {
      return false;
    }
  }
  return true;
}

int isPrimePos(int pos) {
  int position = 0;
  int i = 2;
  while (position < pos) {
    if (isPrime(i)) {
      position++;
      if (position == pos) {
        return i;
      }
    }
    i++;
  }
  return -1;
}

void main() {
  print("Enter the position you want prime no.");
  int pos = int.parse(stdin.readLineSync()!);
  var result = isPrimePos(pos);
  print(result);
}

File: Python/q7.py
def isPrime(number):
    if number <= 1:
        return False
    if number == 2:
        return True
    if number % 2 == 0:
        return False
    for i in range(3, int(number**0.5) + 1, 2):
        if number % i == 0:
            return False
    return True

def pos_Prime(pos):
    position = 0
    i = 2
    while position < pos:
        if isPrime(i):
            position += 1
            if position == pos:
                return i
        i += 1
    return -1

pos = int(input("Enter the position: "))
result = pos_Prime(pos)
print(result)
